ant-networking/src/replication_fetcher.rs

```diff
@@ -323,6 +286,228 @@ impl ReplicationFetcher {
             .collect()
     }
 
+    // Record peers' healthy status after the storage chanllenge.
+    pub(crate) fn add_peer_scores(&mut self, scores: Vec<(PeerId, bool)>) {
+        for (peer_id, is_healthy) in scores {
+            let (peer_scores, last_seen) = self
+                .peers_scores
+                .entry(peer_id)
+                .or_insert((VecDeque::new(), Instant::now()));
+            peer_scores.push_back(is_healthy);
+            if peer_scores.len() > 2 {
+                let _ = peer_scores.pop_front();
+            }
+            *last_seen = Instant::now();
+        }
+
+        // Once got enough scoring knowledge, the `majority` approach shall no longer be used.
+        if self.had_enough_scoring_knowledge() {
+            self.initial_replicates.clear();
+        }
+
+        // Pruning to avoid infinite growing, only keep the recent 20.
+        if self.peers_scores.len() > 20 {
+            let mut oldest_peer = PeerId::random();
+            let mut oldest_timestamp = Instant::now();
+            for (peer_id, (_peer_scores, last_seen)) in self.peers_scores.iter() {
+                if *last_seen < oldest_timestamp {
+                    oldest_timestamp = *last_seen;
+                    oldest_peer = *peer_id;
+                }
+            }
+            let _ = self.peers_scores.remove(&oldest_peer);
```

Comment:
Feels here like this could be exploited by spamming us with newer peer, which would make us forget old pals. 
I'd keep good old peers a bit longer. What do you think?

Reply:
those `newer peers` then can only `replicate in` with majority copies.
replacing old ones with new ones is the most natural way to do.
we could improve it in future anyway, as a non-breaking change to improve the security.

Reply:
Ok make sense